app/controllers/api/v3/sync_controller.rb

```diff
@@ -11,12 +11,14 @@ def __sync_from_user__(params)
   end
 
   def __sync_to_user__(response_key)
-    AuditLog.create_logs_async(current_user, records_to_sync, "fetch", Time.current) unless disable_audit_logs?
+    records = records_to_sync
+
+    AuditLog.create_logs_async(current_user, records, "fetch", Time.current) unless disable_audit_logs?
     render(
-      json: {
-        response_key => records_to_sync.map { |record| transform_to_response(record) },
+      json: Oj.dump({
+        response_key => records.map { |record| transform_to_response(record) },
         "process_token" => encode_process_token(response_process_token)
-      },
+      }, mode: :compat),
```

Comment:
Have you tried using the Oj optimized versions of encoding for some of the core objects? Trying that for dates, times, arrays and hashes would be interesting to see if it helps serialize time.  See https://github.com/ohler55/oj/blob/develop/pages/JsonGem.md ... I think we would need calls like this in an initializer somewhere:

```
Oj.add_to_json(Array)
Oj.add_to_json(Hash)
etc...
```